doc/rvv-intrinsic-spec.adoc

```diff
@@ -316,202 +351,362 @@ The following intrinsics have an irregular naming pattern.
+
 === Assembly generated from the intrinsics
 
-Some users may expect the intrinsics to directly translate and appear in the assembly, the intrinsics are the interfaces that expose the vector instruction semantics. The compiler is free to optimize them out if there is an opportunity.
+Some users may expect the intrinsics to directly translate and appear in the assembly; however, the intrinsics are the interfaces that expose the vector instruction semantics. The implementation is free to optimize them out if there is an opportunity.
 
-=== Bookkeeping of `vtype` in the compiler
+=== Bookkeeping of configurations
 
-The compiler should guarantee that the correct vtype is set given the `EEW` and `EMUL` specified in the intrinsic function name suffix, and the data type of the operand(s).
+Control of `vl`, `vtype`, `vxrm`, and `frm` is not directly exposed to the user. The implementation is responsible for setting the correct values into these CSRs to achieve the expected semantics of the intrinsic functions with respect to the conventions defined in the ISA specification ^0^ and ABI specification ^9^.
 
 === Strided load/store with stride of 0
 
-The V extension spec mentions ^15^ that the strided load/store instruction with stride of 0 could have different behaviors to perform all memory accesses or fewer memory operations. Since needing all memory accesses isn't likely to be common, the compiler implementation is allowed to generate fewer memory operations with strided load/store intrinsics.
+The specification mentions ^15^ that the strided load/store instruction with a stride of 0 could have different behaviors, performing all memory accesses or fewer memory operations. Since needing all memory accesses isn't likely to be common, the implementation is allowed to generate fewer memory operations with strided load/store intrinsics.
+
+In other words, the compiler does not guarantee generating the instruction for all memory accesses in strided load/store intrinsics with a stride of 0. If the user needs all memory accesses to be performed, they should use an indexed load/store intrinsics with all zero indices.
+
+=== Leveraging instructions with operand mnemonics of `vi`
 
-In other words, compiler does not guarantee generating the all memory accesses instruction in strided load/store intrinsics with stride of 0. If the user needs all memory accesses to be performed, they should use an indexed load/store intrinsics with all zero indices.
+The intrinsics provide variants with operand mnemonics of `vv` and `vx`, but not `vi`. This was an intentional design to reduce the total amount of out-going intrinsics. The implementation is responsible to emit instructions with operand mnemonics of `vi` when an immediate that can be expressed within 5-bit is provided to the intrinsics.
```

Comment:
This sounds like a mandatory thing. Like if an implementation had to honour this. Is this the intent?

If this wasn't the intent, maybe rephrasing the last statement to something like "It is a quality of implementation issue whether instructions with `vi` operand mnemonics are emitted when an immediate operand can be encoded in the 5-bit field of the a `vi` instruction".

Reply:
Yes, I think the intent is to not have a `vi` variant and whether or not the compiler is able to recognize 5-bit immediate and use the `vi` instruction is a quality issue.